tests/iface_namingmode/test_iface_namingmode.py

```diff
@@ -960,33 +961,62 @@ def setup_check_topo(self, setup, tbinfo):
 
-    @pytest.fixture(scope='class')
-    def spine_ports(self, setup, tbinfo):
+    @pytest.fixture
+    def static_route_intf(self,  duthosts, enum_rand_one_per_hwsku_frontend_hostname, setup, tbinfo):
         """
         Returns the alias and names of the spine ports
 
         Args:
             setup: Fixture defined in this module
         Returns:
-            spine_ports: dictionary containing lists of aliases and names
+            static_route_intf: dictionary containing lists of aliases and names
             of the spine ports
         """
-        minigraph_neighbors = setup['minigraph_facts']['minigraph_neighbors']
-        spine_ports = dict()
-        spine_ports['interface'] = list()
-        spine_ports['alias'] = list()
-
-        for key, value in list(minigraph_neighbors.items()):
-            if (key in setup['physical_interfaces']
-                    and ('T2' in value['name'] or (tbinfo['topo']['type'] == 't2' and 'T3' in value['name']))):
-                spine_ports['interface'].append(key)
-                spine_ports['alias'].append(setup['port_name_map'][key])
+        duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
+        static_route_intf = dict()
+        static_route_intf['interface'] = list()
+        static_route_intf['alias'] = list()
 
-        if not spine_ports['interface']:
+        if not setup['physical_interfaces']:
             pytest.skip('No non-portchannel member interface present')
 
-        logger.info('spine_ports:\n{}'.format(spine_ports))
-        return spine_ports
+        for mg_intf in setup['minigraph_facts'][u'minigraph_interfaces']:
+            if mg_intf[u'attachto'] == setup['physical_interfaces'][0]:
+                dev = mg_intf[u'attachto']
+                namespace = setup['minigraph_facts']['minigraph_neighbors'][dev]['namespace']
+                gw_ip = mg_intf['peer_addr']
+                if ipaddress.ip_address(gw_ip).version == 4:
+                    ip_version = ''
+                    dst_ip = '192.168.1.1'
+                else:
+                    ip_version = '-6'
+                    dst_ip = '::/0'
+                if namespace:
+                    duthost.shell("ip netns exec {} ip {} route add {}  via {} dev {}".
+                                  format(namespace, ip_version, dst_ip, gw_ip, dev))
+                else:
+                    duthost.shell("ip {} route add {}  via {} dev {}".format(ip_version, dst_ip, gw_ip, dev))
+                static_route_intf['interface'].append(dev)
+                static_route_intf['alias'].append(setup['port_name_map'][dev])
+
+        yield static_route_intf
+        for mg_intf in setup['minigraph_facts'][u'minigraph_interfaces']:
+            if mg_intf[u'attachto'] == setup['physical_interfaces'][0]:
+                dev = mg_intf[u'attachto']
+                namespace = setup['minigraph_facts']['minigraph_neighbors'][dev]['namespace']
```

Comment:
we can avoid this code duplication as mention above with one common function that takes operation type